dev-tools/mage/common.go

```diff
@@ -353,6 +353,77 @@ func unzip(sourceFile, destinationDir string) error {
 	return nil
 }
 
+// Tar compress a directory using tar + gzip algorithms
```

Comment:
Could it be a CLI separate tool?

Reply:
It could be running `tar` directly! 😂 But I wanted to embed it into the build system for cross-platform compatibility

Reply:
cross-platform is super nice, this is something we were talking @v1v and me the other day, this, for example, is one of those cases that we need to do something on all platforms and there is no common way to do that, on each platform we have to install and use similar tools but not the same. To have a goTar CLI tool that only needs Go (directly install the binary from the platform kit from a bucket or something) will simplify our pipelines and build systems.

Reply:
I really like the idea of extracting "things" / commands out to tools. You got me with the unit tests aspect :) 

@jsoriano, wdyt? Can we move on this, or create it a separate tool.

Reply:
Having a set of common CLI tools that work exactly the same in different platforms would be definitely awesome, in many contexts :slightly_smiling_face:

But I also see some potential problems. These tools would become an additional dependency that would need to be installed in CI and in all developers laptops, for newcomers this can suppose an additional thing to handle and learn. Of course this can be overcome by adding some helpers to build tools that transparently install and use the tools, but then this would be another piece to maintain in any new build tool, and maintainers of build scripts would still have the cognitive load of remembering to use these CLI tools instead of the common thing in their platform or language.

If the overhead of supporting this on every build tool is significatively smaller than the overhead of implementing tar and some other common commands in every project where this is needed, then great, but I would be a bit cautious by now. It may happen that [this divergence](https://en.wikipedia.org/wiki/Tar_(computing)#Key_implementations) of behaviours happens in tar but not in so many other common tools.

In summary, I would postpone this for a future and broader conversation.

Reply:
Cool, I think we can merge this one then, and maybe use it for a future follow-up discussion being the case we need to implement those specific tools.